contracts/Governance.sol

```diff
@@ -0,0 +1,480 @@
+    _paramInit();
+
+    require(voteAt == 0 || voteAt >= block.timestamp, "invalid voteAt");
+    if (voteAt == 0) {
+      voteAt = block.timestamp;
+    }
+
+    uint256 totalRequests = requests.length;
+    require(totalRequests > 0, "empty param change request");
+    uint256 endAt = voteAt + votingPeriod;
+
+    Proposal memory proposal;
+    uint256[] memory requestIds = new uint256[](totalRequests);
+    uint256[] memory voteIds = new uint256[](totalRequests);
+
+    for (uint256 i = 0; i < totalRequests; i++) {
+      paramChangeRequests.push(requests[i]);
+      requestIds[i] = requests.length - 1;
+    }
+
+    proposal.requestIds = requestIds;
+    proposal.voteIds = voteIds;
+    proposal.proposer = msg.sender;
+    proposal.description = _description;
+    proposal.startAt = voteAt;
+    proposal.endAt = endAt;
+
+    proposals.push(proposal);
+
+    emit ProposalCreated(proposals.length - 1, msg.sender, _description, voteAt, endAt);
+  }
+
+  function cancelProposal(uint256 proposalId) external {
+    _paramInit();
+
+    Proposal storage proposal = proposals[proposalId];
+    require(msg.sender == proposal.proposer, "only proposer");
+    ProposalState _state = proposalState(proposalId);
+    require(
+      _state == ProposalState.Pending &&
+      _state == ProposalState.Active &&
+      _state == ProposalState.Timelocked &&
+      _state == ProposalState.AwaitingExecution,
+      "invalid proposal state"
+    );
+
+    proposal.canceled = true;
+  }
+
+  function executeProposal(uint256 proposalId) external onlyCabinet {
+    _paramInit();
+
+    require(proposalState(proposalId) == ProposalState.AwaitingExecution, "vote not awaiting execution");
+
+    Proposal storage proposal = proposals[proposalId];
+    proposal.executed = true;
+
+    ParamChangeRequest memory request;
+    for (uint256 i = 0; i < proposal.requestIds.length; i++) {
+      request = paramChangeRequests[proposal.requestIds[i]];
+      IGovHub(GOV_HUB_ADDR).updateParam(request.key, request.value, request.target);
+    }
+
+    emit ProposalExecuted(proposalId, msg.sender);
+  }
+
+  function submitPoll(string calldata description, uint256 voteAt, address shareContract) external {
+    _paramInit();
+    address proposer = msg.sender;
+    ShareLock memory lock = lockShareMap[proposer][shareContract];
+    require(lock.votingPower >= pollSubmitThreshold, "locked voting power not enough");
+
+    require(voteAt == 0 || voteAt >= block.timestamp, "invalid voteAt");
+    if (voteAt == 0) {
+      voteAt = block.timestamp;
+    }
+
+    uint256 endAt = voteAt + pollVotingPeriod;
+    Poll memory poll;
+    poll.proposer = proposer;
+    poll.description = description;
+    poll.startAt = voteAt;
+    poll.endAt = endAt;
+
+    polls.push(poll);
+
+    emit PollCreated(polls.length - 1, msg.sender, description, voteAt, endAt);
+  }
+
+  function lockShare(address shareContract, uint256 shareAmount) external {
```

Comment:
I think `lockShare` and `unlockShare` is unnecessary here, I believe `castVote` is actually doing the lock stBNB